src/Symfony/Component/Serializer/Encoder/XmlEncoder.php

```diff
@@ -292,16 +292,17 @@ private function parseXml(\DOMNode $node)
      *
      * @return array
      */
-    private function parseXmlAttributes(\DOMNode $node)
+    private function parseXmlAttributes(\DOMNode $node, array $context = array())
     {
         if (!$node->hasAttributes()) {
             return array();
         }
 
         $data = array();
+        $typeCastAttributes = $this->resolveXmlTypeCastAttributes($context);
 
         foreach ($node->attributes as $attr) {
-            if (!is_numeric($attr->nodeValue)) {
+            if (!is_numeric($attr->nodeValue) || !$typeCastAttributes) {
```

Comment:
@nicolas-grekas That sounds good. That would fix my specific use case, not sure if it fixes *all* of the issues others might have.

I might just submit another MR then with the bug fix like @nicolas-grekas shows into 2.7. And rebase this off of 3.4